### PR TITLE
* Switch from setup.py to pyproject.toml

### PR DESCRIPTION
**Pull request by mkopec87**
* Switch from setup.py to pyproject.toml
* Add numpy<2,pandas<2 test environment to build pipeline test matrix

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: Apache Software License",
+    "Topic :: Scientific/Engineering :: Information Analysis",
+    "Topic :: Scientific/Engineering :: Mathematics",
+    "Topic :: Scientific/Engineering :: Physics",
+]
+dynamic = ["version"]
+
+[project.optional-dependencies]
+test = [
```
Comment (mbaak) [inline pyproject.toml:39]: Can you add the optional dependencies:
`pandas = [
    "pandas"
]

spark = [
  "pyspark>=3.1; python_version <= '3.11'",
]
`

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -9,27 +9,37 @@ jobs:
...
+          restore-keys: |
+            ${{ runner.os }}-pip-
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          if [ "${{ matrix.numpy_version }}" = "numpy<2" ]; then
+            pip install ".[test,test_numpy_pre2]"
+          else
+            pip install ".[test]"
```
Comment (mbaak) [inline .github/workflows/test.yml:40]: Can you switch `[test]` -> `[test,spark]` ?
(See also pyproject.toml below).
I _think_ that will trigger the spark tests.

File: pyproject.toml (hunk 1 above)
Comment (mkopec87) [reply pyproject.toml:45]: Done

File: .github/workflows/test.yml (hunk 2 above)
Comment (mkopec87) [reply .github/workflows/test.yml]: Added, seems it works :)

However, should we also add "spark" to "numpy<2" test?
```pip install ".[test,spark,test_numpy_pre2]"```

**Review by mbaak — APPROVED** [review]

**Pull request merged**